### PR TITLE
Allows you to tear out vaurca mandibles

### PR DESCRIPTION
**Pull request by LadyFowls**
Allows you to tear out a new organ, vaurca mandibles, preventing them from biting you

### CODE REVIEW

**Comment by LadyFowls** [comment]
!review

File: code/game/objects/items/weapons/vaurca_items.dm [hunk 1]
```diff
@@ -116,6 +116,7 @@
 		H.update_inv_r_hand()
 	to_chat(user, "<span class='notice'>\The [src] is de-energised.</span>")
 
+
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/vaurca_items.dm:119]: Rogue space.

File: code/modules/mob/mob_grab_specials.dm [hunk 2]
```diff
@@ -202,3 +202,61 @@
 				visible_message("<span class='danger'>[assailant] violently tugs [target]'s [hairchatname]!</span>")
 				target.apply_damage(15, PAIN)
 				src.state = GRAB_AGGRESSIVE
+
```
Comment (NonQueueingMatt) [inline code/modules/mob/mob_grab_specials.dm:205]: Rogue space.

File: code/modules/mob/mob_grab_specials.dm [hunk 3]
```diff
@@ -202,3 +202,61 @@
...
+
+/obj/item/grab/proc/mandible_pull(mob/living/carbon/human/target, mob/living/carbon/human/attacker, var/target_zone)
+	var/obj/item/organ/vaurca/mandibles/O = target.internal_organs_by_name[BP_VAURCA_MANDIBLE]
+	if(isunathi(attacker) || (istype(attacker.gloves,/obj/item/clothing/gloves/force)))
+		switch(O.mandible_size)
+			if(1)
```
Comment (NonQueueingMatt) [inline code/modules/mob/mob_grab_specials.dm:211]: Magic numbers, should probably be defines.

File: code/modules/mob/mob_grab_specials.dm [hunk 4]
```diff
@@ -202,3 +202,61 @@
...
+
+
+/obj/item/grab/proc/mandible_pull(mob/living/carbon/human/target, mob/living/carbon/human/attacker, var/target_zone)
+	var/obj/item/organ/vaurca/mandibles/O = target.internal_organs_by_name[BP_VAURCA_MANDIBLE]
+	if(isunathi(attacker) || (istype(attacker.gloves,/obj/item/clothing/gloves/force)))
+		switch(O.mandible_size)
+			if(1)	
+				if(prob(45))
+					visible_message("<span class='danger'>[assailant] violently tugs on [target]'s mandible, ripping them out!</span>")
+					target.apply_damage(75, PAIN)
```
Comment (NonQueueingMatt) [inline code/modules/mob/mob_grab_specials.dm:214]: You can directly add the pain to the head organ here.

File: code/modules/mob/mob_grab_specials.dm [hunk 5]
```diff
@@ -202,3 +202,61 @@
...
+					O.forceMove(target.loc)
+					O.removed(target, attacker)
+					qdel(src)
+					attacker.put_in_hands(O)
+					target.verbs -= /mob/living/carbon/human/proc/bugbite
+					to_chat(target, "<span class='warning'>Your mandible is torn out!</span>")
+					new /obj/effect/gibspawner/xeno(target.loc)
+				else
+					visible_message("<span class='danger'>[assailant] violently tugs on [target]'s mandible, but slices their hand in the process and lets go!</span>")
+					target.apply_damage(15, PAIN)
```
Comment (NonQueueingMatt) [inline code/modules/mob/mob_grab_specials.dm:225]: ditto

File: code/modules/mob/mob_grab_specials.dm [hunk 6]
```diff
@@ -202,3 +202,61 @@
...
+					attacker.apply_damage(5, BRUTE)
+					src.state = GRAB_PASSIVE
+			if(2)	
+				if(prob(25))
+					visible_message("<span class='danger'>[assailant] violently tugs on [target]'s mandible, ripping them out!</span>")
+					target.apply_damage(75, PAIN)
```
Comment (NonQueueingMatt) [inline code/modules/mob/mob_grab_specials.dm:231]: ditto

File: code/modules/organs/subtypes/vaurca.dm [hunk 7]
```diff
@@ -76,6 +76,28 @@ obj/item/organ/vaurca/neuralsocket/process()
...
+
+/obj/item/organ/vaurca/mandibles
+	var/mandible_size = 1 // This is just for safety so we don't have to do a bunch of dumb species checks. Can't have people easily just tearing out a warforms mandibles
+	name = BP_VAURCA_MANDIBLE
+	icon = 'icons/obj/vaurca_items.dmi'
+	organ_tag = BP_VAURCA_MANDIBLE
+	icon_state = "mandible"
+	parent_organ = BP_HEAD
+	sharp = 1
+	edge = 1
```
Comment (NonQueueingMatt) [inline code/modules/organs/subtypes/vaurca.dm:88]: booleans

File: code/modules/organs/subtypes/vaurca.dm [hunk 8]
```diff
@@ -76,6 +76,28 @@ obj/item/organ/vaurca/neuralsocket/process()
...
+
+/obj/item/organ/vaurca/mandibles/warform
+	mandible_size = 2
+	name = BP_VAURCA_MANDIBLE
+	organ_tag = BP_VAURCA_MANDIBLE
+	icon_state = "mandible_warform"
+	parent_organ = BP_HEAD
+	sharp = 1
+	edge = 1
+	hitsound = 'sound/weapons/bladeslice.ogg'
```
Comment (NonQueueingMatt) [inline code/modules/organs/subtypes/vaurca.dm:99]: this is a subtype, you don't need most of these variables to be re-defined

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by NonQueueingMatt** [comment]
Inactive author.

**Pull request closed without merging**